doc/user-manual/tools/command-line-options.rst

```diff
@@ -476,6 +476,21 @@ Other features
      Disable the syntactic equality shortcut in the conversion
      checker.
 
+.. option:: --syntactic-equality={N}
+
+     .. versionadded:: 2.6.3
+
+     Give the syntactic equality shortcut ``N`` units of fuel (``N``
+     must be a natural number).
+
+     If ``N`` is omitted, then the syntactic equality shortcut is
+     enabled without any restrictions.
+
+     If ``N`` is given, then the syntactic equality shortcut is given
+     ``N`` units of fuel. The exact meaning of this is
+     implementation-dependent, but successful uses of the shortcut do
```

Comment:
For a user-manual, I still find this explanation mystifying.  Documentation should not be [hermetic poetry](https://en.wikipedia.org/wiki/Hermeticism_(poetry)).

Reply:
I've already explained why I did not include more details:

> I did not include this explanation in the user manual, because it is tied to the implementation, and I do not know the details of the implementation well enough to give a more high-level explanation.

Reply:
Well, if it is too complicated to explain how it works, then there could be some advice instead when its use may be beneficial or could be attempted.
The user manual addresses users, so it should have documentation that can be understood by users.
To me, the sentence _successful uses of the shortcut do not affect the amount of fuel_ is hermetic.
And I do not understand in which situations it may help me to give the option `--syntactic-equality=100` or so.

Reply:
I added this as an experiment, after reading about the implementation of smalltt (see #5801). I constructed one test case where the use of syntactic equality makes Agda very slow ([`test/Succeed/Issue5801.agda`](https://github.com/agda/agda/blob/8b7c6d886b835f5dfcf1e5052dd83ebabac73592/test/Succeed/Issue5801.agda)). This test case seems to run a little faster with `--syntactic-equality=2` than with `--no-syntactic-equality`. For the standard library the situation is [different](https://github.com/agda/agda/issues/5801): `--syntactic-equality=24` seems to be comparable with `--syntactic-equality`, and `--no-syntactic-equality` is worse (at least that was the case when I ran my experiments, things may have changed). If you were to add my artificial test case to the standard library, then it might make sense to use, say, `--syntactic-equality=24`. However, that test case is artificial, and I am not aware of any "real" pieces of code where it makes sense to use `--syntactic-equality=N` for positive `N`.

Given your complaints about the documentation, and the lack of a killer application for the new option, I suggest that we remove the documentation, and hide the option from `--help`. (I don't want to remove the option entirely, because I want to see if things change if we borrow more ideas from smalltt.)

Reply:
@andreasabel, do you think that we should remove the documentation as I suggest above?

Reply:
You could mark this option as _experimental_ to indicate it might be subject to changes.